### PR TITLE
Making changes necessary for allow for cross compilation

### PR DESCRIPTION
**Pull request by tylerhawkes**
This should all most cross-compiling to work. I've tested it for aarch64 and it compiles into a binary just fine. When not cross-compiling it still works the same.

I changed some unwrap() calls to expect() to make it easier to diagnose failures as well.

### CODE REVIEW

File: build.rs [hunk 1]
```diff
@@ -199,3 +211,67 @@ fn main() {
...
+//+      -Wl,--as-needed
+//+  endif
+//+
+//+  LIBS := \
+//+    -lstdc++ \
+//+    -lpthread \
+//+    -lm \
+//+    -ldl
+//+
+//+endif
```
Comment (boncheolgu) [inline build.rs]: could you explain what this is for?

Comment (tylerhawkes) [reply build.rs]: I was using a known good implementation from another tensorflow branch to get the build script working. I've removed the comments and added some extra comments to make it clear why the changes are necessary.

**Comment by tylerhawkes** [comment]
When this is ready to merge, please just squash it.

**Comment by boncheolgu** [comment]
looks great, thx!

**Pull request merged**